Bindings/Java/OpenSimJNI/OpenSimContext.cpp

```diff
@@ -220,19 +223,21 @@ void OpenSimContext::setRangeMax(ConditionalPathPoint&  via, double d) {
     return;
 }
 
-bool OpenSimContext::replacePathPoint(GeometryPath& p, PathPoint& mp, PathPoint& newPoint) {
+bool OpenSimContext::replacePathPoint(GeometryPath& p, AbstractPathPoint& mp, AbstractPathPoint& newPoint) {
    bool ret= p.replacePathPoint(*_configState, &mp, &newPoint );
    recreateSystemAfterSystemExists();
    realizeVelocity();
    p.updateGeometry(*_configState);
    return ret;
 }
 
-void OpenSimContext::setLocation(PathPoint& mp, int i, double d) {
-    mp.setLocation(*_configState, i, d);
-  _configState->invalidateAll(SimTK::Stage::Position);
-  _model->getMultibodySystem().realize(*_configState, SimTK::Stage::Position);
-    return;
+void OpenSimContext::setLocation(AbstractPathPoint& mp, int i, double d) {
+    PathPoint* spp = dynamic_cast<PathPoint*>(&mp);
+    if (spp) {
+        spp->setLocationCoord(*_configState, i, d);
+        _configState->invalidateAll(SimTK::Stage::Position);
+        _model->getMultibodySystem().realize(*_configState, SimTK::Stage::Position);
+    }
```

Comment:
If we don't want "to do nothing", which was effectively what was happening before, I would prefer to only allow `OpenSim::setLocation(PathPoint ...)` and make it impossible to call with a `MovingPathPoint` altogether.